app/src/main/java/com/avans/rentmycar/ui/mycars/CarAddItemFragment.kt

```diff
@@ -50,33 +57,68 @@ class CarAddItemFragment : Fragment() {
 
-        binding.buttonGetLicenseplateRdw.setOnClickListener {
+        binding.buttonCarGetRdwdetails.setOnClickListener {
             Log.d(TAG, "onViewCreated() => Button GET clicked. Invoke RdwApiService")
-            val kenteken: TextInputEditText = binding.txtInputLicensePlate
+            val kenteken: TextInputEditText = binding.txtInputCarLicensePlate
             val licensePlate: String = kenteken.text.toString()
 
             // Validate licenseplate before API call
             if (isValidLicensePlate()){
                 // invoke RdwApiService for retrieval of cardetails of the given licenseplate
-                Log.d(TAG, "onViewCreated() => invoke RdwApieService for licenseplate: " + licensePlate)
-                carAddItemViewModel.getRdwCarDetails(kenteken.text.toString())
+                Log.d(TAG, "onViewCreated() => invoke RdwApiService for licenseplate: " + licensePlate)
+                carViewModel.getRdwCarDetails(kenteken.text.toString())
                 Snackbar.make(view, "Car details retrieved at the RDW", Snackbar.LENGTH_LONG)
                 .show()
             }
         }
 
-        // Observer for rdwResponse
-        Log.d(TAG, "onViewCreated() => set observer on rdw response ")
-        carAddItemViewModel.rdwResponse.observe(viewLifecycleOwner) {
+        binding.buttonCarSave.setOnClickListener {
+            Log.d(TAG, "onViewCreated() => Button SAVE clicked. Invoke CarApiService")
+            // TODO Add validation for all inputfield
+            createCar()
+            Log.d(TAG, "onViewCreated() => After calling fun createCar()")
+        }
+
+        carViewModel.rdwResponse.observe(viewLifecycleOwner) {
+            Log.d(TAG, "onViewCreated() => observer rdwResponse triggerd")
+            if (carViewModel.rdwResponse.value!!.isEmpty()){
+                Snackbar.make(view, "Error saving Car. Please try again later", Snackbar.LENGTH_LONG)
+                    .show()
+                }
             bindUI(it)
         }
+
+        carViewModel.carCreateResponse.observe(viewLifecycleOwner) {
+            Log.d(TAG, "onViewCreated() => observer carCreateResponse triggerd")
+            when (it) {
+                is BaseResponse.Loading -> {
+                    Log.d(TAG, "onViewCreated() => observer BaseResponse.Loading")
+
+                }
+                is BaseResponse.Error -> {
+                    Toast.makeText(context, "Error", Toast.LENGTH_LONG).show()
+                    Snackbar.make(view, "An error occurred while fetching your data. Please try again later", Snackbar.LENGTH_LONG)
+                            .show()
+                    Log.d(TAG, "onViewCreated() => observer BaseResponse.Error")
+
+                }
+                is BaseResponse.Success -> {
+                    Log.d(TAG, "onViewCreated() => observer BaseResponse.Success. Return to MyCarsFragment ")
+                    val action = CarDetailFragmentDirections.actionCarDetailFragmentToMycars()
```

Comment:
De auto wordt toegevoegd, maar ik krijg wel een error in logcat en ik navigeer niet terug
`createCar() => Exception: Error(msg=Navigation action/destination com.avans.rentmycar:id/action_carDetailFragment_to_mycars cannot be found from the current destination Destination(com.avans.rentmycar:id/carAddItemFragment) label=Add a car class=com.avans.rentmycar.ui.mycars.CarAddItemFragment)`

Reply:
Heb hier een isasue voor aangemaakt. Geldt bij elke CRUD actie op een Car